test/DaprExtensionTests/Tests/DaprTopicTriggerTests.cs

```diff
@@ -92,31 +94,46 @@ public async Task InvokeSubscriptionEndpoint()
-            // Version 0.8 only support route to match topic name
-            IEnumerable<string> routes = array.Select(item => (JObject)item).Select(obj => (string)obj.GetValue("route"));
-            Assert.Contains(nameof(Functions.IntTopic), routes);
-            Assert.Contains(nameof(Functions.CustomTypeTopic), routes);
-            Assert.Contains(nameof(Functions.StringTopic), routes);
-            Assert.Contains(nameof(Functions.StreamTopic), routes);
-            Assert.Contains(nameof(Functions.BytesTopic), routes);
-            Assert.Contains(nameof(Functions.JObjectTopic), routes);
-            Assert.Contains(nameof(Functions.CloudEventTopic), routes);
-
-            // Make sure the explicit topic names are handled correctly
-            Assert.Contains("MyTopic", topics);
-            Assert.DoesNotContain(nameof(Functions.DotNetMethodName), topics);
-            Assert.DoesNotContain("MyFunctionName", topics);
+            var subscriptions = array
+                .Cast<JObject>()
+                .Select(obj => (name: (string)obj.GetValue("pubsubname"), topic: (string)obj.GetValue("topic"), route: (string)obj.GetValue("route")))
+                .OrderBy(t => t.topic)
+                .ToArray();
+
+            Assert.Collection(
+                subscriptions,
+                s => AssertDefaults(s, nameof(Functions.BytesTopic)),
+                s => AssertDefaults(s, nameof(Functions.CloudEventTopic)),
+                s => AssertDefaults(s, nameof(Functions.CustomTypeTopic)),
+                s => AssertDefaults(s, nameof(Functions.IntTopic)),
+                s => AssertDefaults(s, nameof(Functions.JObjectTopic)),
+                s =>
+                {
+                    // This one has a custom configuration with env-vars
+                    Assert.Equal("MyBoundPubSub", s.name);
+                    Assert.Equal("MyBoundTopic", s.topic);
+                    Assert.Equal("MyBoundTopic", s.route);
+                },
+                s =>
+                {
+                    // This one has a custom configuration
+                    Assert.Equal("MyOtherPubSub", s.name);
+                    Assert.Equal("MyTopic", s.topic);
+                    Assert.Equal("MyRoute", s.route);
+                },
+                s => AssertDefaults(s, nameof(Functions.StreamTopic)),
+                s => AssertDefaults(s, nameof(Functions.StringTopic)));
+
+            Assert.DoesNotContain(nameof(Functions.DotNetMethodName), subscriptions.Select(s => s.topic));
+            Assert.DoesNotContain("MyFunctionName", subscriptions.Select(s => s.topic));
+
+            void AssertDefaults((string name, string topic, string route) s, string methodName)
+            {
+                // by default the method name is the topic and the route
+                Assert.Equal("MyPubSub", s.name);
+                Assert.Equal(methodName, s.topic);
+                Assert.Equal(methodName, s.route); 
+            }
```

Comment:
This is a slightly less error prone way to write these tests. If you add more cases, you have to add more lines. The previous approach would have missed "extra" stuff.

Reply:
good catch